File: app/src/main/java/com/esorokin/lantector/di/module/AndroidApplicationModule.java
package com.esorokin.lantector.di.module;

import android.content.Context;

import com.esorokin.lantector.app.StringProvider;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AndroidApplicationModule {
	private final Context context;

	public AndroidApplicationModule(Context context) {
		this.context = context;
	}

	@Provides
	@Singleton
	Context provideContext() {
		return context;
	}

	@Provides
	@Singleton
	StringProvider provideStringProvider() {
		return context::getString;
	}
}

File: app/src/main/java/com/esorokin/lantector/di/module/AndroidApplicationModule.kt
package com.esorokin.lantector.di.module

import android.content.Context
import com.esorokin.lantector.app.StringProvider
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AndroidApplicationModule(private val context: Context) {
    @Provides
    @Singleton
    fun provideContext(): Context {
        return context
    }

    @Provides
    @Singleton
    fun provideStringProvider(): StringProvider {
        return object : StringProvider {
            override fun getString(stringRes: Int, vararg formatArgs: Any): String = context.getString(stringRes, formatArgs)
        }
    }
}
